provisioning/security/security-provider/src/main/java/com/microsoft/azure/sdk/iot/provisioning/security/SecurityProviderX509.java

```diff
@@ -27,7 +27,7 @@
     abstract public Collection<X509Certificate> getIntermediateCertificatesChain();
 
     @Override
-    public String getRegistrationId() throws SecurityProviderException
```

Comment:
This would also be a breaking change since users do call this method directly under some circumstances